Modules/_sre/sre.c

```diff
@@ -2614,6 +2677,89 @@ _sre_SRE_Match_span_impl(MatchObject *self, PyObject *group)
     return _pair(self->mark[index*2], self->mark[index*2+1]);
 }
 
+/*[clinic input]
+_sre.SRE_Match.index
+
+    value: object
+    start: slice_index(accept={int}) = 0
+    stop: slice_index(accept={int}, c_default="PY_SSIZE_T_MAX") = sys.maxsize
+    /
+
+Return the index of the first occurrence of the value among the matched groups.
+
+Raises ValueError if the value is not present.
+[clinic start generated code]*/
+
+static PyObject *
+_sre_SRE_Match_index_impl(MatchObject *self, PyObject *value,
+                          Py_ssize_t start, Py_ssize_t stop)
+/*[clinic end generated code: output=846597f6f96f829c input=7f41b5a99e0ad88e]*/
+{
+    Py_ssize_t i;
+
+    if (start < 0) {
+        start += self->groups;
+        if (start < 0) {
+            start = 0;
+        }
+    }
+    if (stop < 0) {
+        stop += self->groups;
+    }
+    else if (stop > self->groups) {
+        stop = self->groups;
+    }
+    for (i = start; i < stop; i++) {
+        PyObject* group = match_getslice_by_index(self, i, Py_None);
+        if (group == NULL) {
+            return NULL;
+        }
+        int cmp = PyObject_RichCompareBool(group, value, Py_EQ);
+        Py_DECREF(group);
+        if (cmp > 0) {
+            return PyLong_FromSsize_t(i);
+        }
+        else if (cmp < 0) {
+            return NULL;
+        }
```

Comment:
Let's handle the error value before (even though it doesn't matter here but it's better to first handle errors)